### PR TITLE
Style fixes for secondary_axis.

### PR DESCRIPTION
**Pull request by anntzer**
There's a lot of unused imports :)

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/axes/_secondary_axes.py [hunk 1]
```diff
@@ -1,30 +1,11 @@
...
-    FuncFormatter,
-    LogFormatterSciNotation,
-    LogLocator,
-    NullLocator,
-    NullFormatter,
-    ScalarFormatter
-)
-
-from matplotlib.scale import Log10Transform
+from matplotlib.ticker import FixedLocator
```
Comment (afvincent) [inline lib/matplotlib/axes/_secondary_axes.py]: Why not leverage (the already imported) `mticker`? AFAICT, only line 273 would need to be updated from
```
self._axis.set_major_locator(FixedLocator(ticks))
```
to
```
self._axis.set_major_locator(mticker.FixedLocator(ticks))
```

File: lib/matplotlib/axes/_secondary_axes.py [hunk 2]
```diff
@@ -254,9 +235,7 @@ def draw(self, renderer=None, inframe=False):
         using the converter specified by
         `~.axes._secondary_axes.set_functions` (or *functions*
         parameter when axes initialized.)
-
         """
```
Comment (afvincent) [inline lib/matplotlib/axes/_secondary_axes.py]: By curiosity, is there a reason for removing the blank line after the docstring? It looks like the file already hosts a mix of “blank line”/“no blank line” after a docstring anyway…

File: lib/matplotlib/axes/_secondary_axes.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/axes/_secondary_axes.py]: done

File: lib/matplotlib/axes/_secondary_axes.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/axes/_secondary_axes.py]: See https://github.com/PyCQA/pydocstyle/issues/361 for how I tend to use blank lines (basically, use one if the body has other blank lines).  Not religious about that, but given that you asked, I made the changes throughout the module :)

**Pull request merged**